### PR TITLE
Replace `try` with `lookup`

### PR DESCRIPTION
**Pull request by lawliet89**
See https://github.com/clowdhaus/terraform-for-each-unknown

Also, allow service specific IAM roles

Add support for _no_ task IAM role.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -42,32 +42,33 @@ module "cluster" {
...
-  task_cpu              = each.value.service_task_cpu
-  task_memory           = each.value.service_task_memory
-  desired_count         = each.value.service_desired_count
-  execution_role_arn    = var.service_task_execution_role_arn
-  task_role_arn         = var.service_task_role_arn
+  task_cpu              = lookup(each.value, "service_task_cpu", null)
+  task_memory           = lookup(each.value, "service_task_memory", null)
+  desired_count         = lookup(each.value, "service_desired_count", null)
+  execution_role_arn    = lookup(each.value, "execution_role_arn", var.service_task_execution_role_arn)
+  task_role_arn         = lookup(each.value, "task_role_arn", var.service_task_role_arn)
```
Comment (lawliet89) [inline main.tf:57]: Actually, these should be mandatory right?

Comment (niroz89) [reply main.tf:57]: yes. need to check which are, container_definition for sure mandatory.

https://github.com/SPHTech-Platform/terraform-aws-ecs/pull/3#discussion_r1007557654

Comment (niroz89) [reply main.tf:57]: task_cpu and task_memory are required for ecs fargate.

Comment (lawliet89) [reply main.tf]: Changed some of them. Good practice to always define task cpu and memory anyway

**Review by niroz89 — APPROVED** [review]

**Review by wayne-root — APPROVED** [review]
LGTM

**Pull request merged**